mesonbuild/envconfig.py

```diff
@@ -132,6 +132,7 @@
     'pkg-config': 'PKG_CONFIG',
     'make': 'MAKE',
     'vapigen': 'VAPIGEN',
+    'llvm_config': 'LLVM_CONFIG'
```

Comment:
The program name has a `-`, not a `_`.

```suggestion
    'llvm-config': 'LLVM_CONFIG'
```

This would allow `find_program('llvm-config')` to work as expected, and also vastly simplifies the rest of this PR.

Reply:
Ahh, didn't realize find_program could find the program with the right program name to environment variable mapping. Changed.